protocol/tx/entry.go

```diff
@@ -73,6 +97,10 @@ func writeForHash(w io.Writer, c interface{}) error {
 	case bc.AssetID:
 		_, err := w.Write(v[:])
 		return errors.Wrap(err, "writing bc.AssetID for hash")
+
+	case *EntryRef:
```

Comment:
Ahh, I was trying to suggest something quite different. Should we move the pointers — not the hashes — out of the body in each concrete entry type? For example:

```go
type issuance struct {
	body struct {
		Anchor  bc.Hash
		Value   bc.AssetAmount
		Data    bc.Hash
		ExtHash bc.Hash
	}
	ordinal int
	anchor  entry // equiv to body.Anchor
}
```

or

```go
type spend struct {
	body struct {
		SpentOutput bc.Hash // must be an Output entry
		Data        bc.Hash
		ExtHash     bc.Hash
	}
	ordinal     int
	spentOutput *output // equiv to body.SpentOutput
}
```

etc